src/lib/y2storage/storage_manager.rb

```diff
@@ -40,14 +41,13 @@ module Y2Storage
   # there are quite some code that could be extracted to a new place, mainly all
   # stuff related to testing (e.g., {#probe_from_yaml}).
   #
-  class StorageManager
+  class StorageManager # rubocop:disable ClassLength
```

Comment:
hmm, this show some design problems with this class. It kind of becoming god object, which is not good. Also file is already 700 lines long, which is quite lot.

Quickly going over this class few hints what need not to be in this class:

1. logger and its setting. It should be separate file like e.g. logger for cheetah - https://github.com/yast/yast-yast2/blob/66bc64d08c6ff017b88e3cff63fd02d78bbcf7f1/library/system/src/lib/yast2/execute.rb#L43 so it lives in its own file and you just need to require it.

2. create_test_instance. It can live outside of production code. Ideally in tests in separate file.

3. configure_snapper? feels a bit meh in StorageManager. I expect it in some kind of proposal or installation specific class.

4. probe_from_* feels a bit strange as I expect to be done outside and just assign probe graph to manager

In general issue of this class is that it has too big scope, from low level operations like commit, access to probed, staging, probe to quite high level operations.

So I prefer to ideally address it or at least add there TODO with plan how to handle this situation.

Reply:
Totally agree. In fact, this class and the filesystem controller deserve a refactoring card. @ancorgs is there any plan for that?

Anyway, I would address it with a separate card.

Reply:
No concrete card for `StorageManager` so far, but feel free to create one.

Reply:
Ok, I have created this card: https://trello.com/c/Tg1UvmmZ/385-storagemanager-class-needs-refactoring